### PR TITLE
add stable/xwiki chart

### PR DESCRIPTION
**Pull request by flypenguin**
#### What this PR does / why we need it:
Add `stable/xwiki` chart:
- https://www.xwiki.org
- https://hub.docker.com/_/xwiki

#### Which issue this PR fixes
None

#### Special notes for your reviewer:
None

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by Miouge1** [comment]
Looks good to me.

**Comment by flypenguin** [comment]
@Miouge1 - what happens now? Can I do something?

**Comment by Miouge1** [comment]
Only folks from https://github.com/helm/charts/blob/master/OWNERS can get PR merged, try to ping some of them.

File: stable/xwiki/templates/statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    app.kubernetes.io/name: {{ include "xwiki.name" . }}
+    helm.sh/chart: {{ include "xwiki.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  # without shared storage, no more than 1 replica, also we can't start the next
+  # before killing the old due to the volume attachment. I think.
+  # anyway, safer this way.
+  replicas: 1
+  strategy:
```
Comment (cristi-) [inline stable/xwiki/templates/statefulset.yaml]: According to the spec, for StatefulSets this should be `updateStrategy` ?

File: stable/xwiki/templates/statefulset.yaml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    type: Recreate
```
Comment (cristi-) [inline stable/xwiki/templates/statefulset.yaml]: This can only be `RollingUpdate` or `onDelete`

File: stable/xwiki/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  # ... aaand continue as usual.
```
Comment (cristi-) [inline stable/xwiki/templates/statefulset.yaml]: You should add a `serviceName`, that's required for statefulsets.

File: stable/xwiki/values.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+db:
+  user: xwiki
+  pass: xwiki
+  host: localhost
+  database: xwiki
+
+persistence:
+  enabled: false
+  size: 1GiB
```
Comment (cristi-) [inline stable/xwiki/values.yaml]: This should be 1G

Comment (flypenguin) [reply stable/xwiki/values.yaml]: https://is.gd/JtAEMn says I am wrong but I can still use "Gi" :) . I'd like to use the power-of-two values, cause it's more binary.

File: stable/xwiki/templates/statefulset.yaml (hunk 1 above)
Comment (flypenguin) [reply stable/xwiki/templates/statefulset.yaml]: corrected.

File: stable/xwiki/templates/statefulset.yaml (hunk 2 above)
Comment (flypenguin) [reply stable/xwiki/templates/statefulset.yaml]: true ... not sure where I got that.

File: stable/xwiki/templates/statefulset.yaml (hunk 3 above)
Comment (flypenguin) [reply stable/xwiki/templates/statefulset.yaml]: added.

File: stable/xwiki/values.yaml (hunk 4 above)
Comment (flypenguin) [reply stable/xwiki/values.yaml]: https://is.gd/JtAEMn says I am wrong but apparently I can still use "Gi" :) . I'd like to use the power-of-two values, cause it's more binary.

**Comment by flypenguin** [comment]
@cristi- ping. merged your changes, not *entirely* sure about the "Gi" thing but it seems valid. Personally I like base-2 more than base-10 size values. I'd appreciate another review.

**Comment by maorfr** [comment]
hey @flypenguin 

Have you considered hosting this chart and adding it to helm hub?
https://github.com/helm/hub/blob/master/Repositories.md

**Comment by flypenguin** [comment]
@maorfr - I'd like to do this. still I want to be included here as well. ;)
/ping @cristi-

**Comment by flypenguin** [comment]
or maybe also ping @viglesiasce / @foxish :)

**Comment by maorfr** [comment]
I think that you totally should host your own charts!
This will give you more freedom around management, and you will not be blocked by us (maintainers).

This means that when users want to install your chart, instead of installing them from stable, they will install it from (for example) mycompany-charts:

$ helm install --name my-release mycompany-charts/mychart
Here are some instructions about how to host your own charts repo:
https://github.com/helm/hub/blob/master/docs/github.md#github-pages-as-a-repository

In order for people to find out about your charts repo, you should add it to helm hub:
https://hub.helm.sh/

Here are some instructions about how to add your charts repo to helm hub:
https://github.com/helm/hub#how-to-add-your-helm-charts

and just as a note, it is better not to have the same charts in different repositories. so, if you intend to self host - it would be better not to add it here.

wdyt?

**Comment by flypenguin** [comment]
@maorfr so what are you saying? I should *not* try to be in stable/ ? cause I myself always look in charts/stable if I want to find a chart. it's kinda the go-to place, and I want xwiki to be in there so that most people can find it quickly.

**Comment by maorfr** [comment]
check this out:
https://helm.sh/blog/intro-helm-hub/

we are moving towards helm hub. better to be there, as all eyes will eventually be there and not here...

**Comment by flypenguin** [comment]
so @maorfr, sorry to be a pain, but you *are* saying "drop this here and move over to helm hub", right? I personally don't like it that much, cause right now I guess you'd lose `helm search`, visibility, and a couple of other things.

I'll look into this though, it's just I'm **fully done** here just missing a review, and I have to read up on the hub. (yes it might be short but it's more than nothing ...)

**Comment by maorfr** [comment]
I'm not saying that :)

You are more then welcome to keep this PR open and wait until one of the maintainers has time to review.
I would say that this is not done since this chart is not self sufficient:
https://github.com/helm/charts/pull/11089/files#diff-22e29f675342ee2f963ed44fd2df04c1R7

Plus, you would not lose `helm search`, but users will have to know your repository.

Please, feel free to keep this PR open for review. I do suggest that you try to look into helm hub in parallel :)

thanks for contributing!

**Comment by flypenguin** [comment]
> I would say that this is not done since this chart is not self sufficient:
> https://github.com/helm/charts/pull/11089/files#diff-22e29f675342ee2f963ed44fd2df04c1R7

what?! I changed that! where did I lose this? argh.

> Plus, you would not lose `helm search`, but users will have to know your repository.

yeah, that's the issue ;)

> Please, feel free to keep this PR open for review. I do suggest that you try to look into helm hub in parallel :)

will do!

**Pull request closed without merging**